Firestore/core/src/firebase/firestore/model/field_path.cc

```diff
@@ -59,6 +59,23 @@ bool IsValidIdentifier(const std::string& segment) {
   return true;
 }
 
+/** A custom formatter to be used with absl::StrJoin(). */
+struct JoinEscaped {
+  static std::string escaped_segment(const std::string& segment) {
+    auto escaped = absl::StrReplaceAll(segment, {{"\\", "\\\\"}, {"`", "\\`"}});
+    const bool needs_escaping = !IsValidIdentifier(escaped);
+    if (needs_escaping) {
+      escaped.insert(escaped.begin(), '`');
+      escaped.push_back('`');
+    }
+    return escaped;
+  }
+
+  template <typename T>
+  void operator()(T* out, const std::string& segment) {
```

Comment:
Does this need to be templated? i.e. would s/T* out/std::string* out/ work?

Reply:
This is to resolve the conflict of `std::string` and `string`. This way, both 3p and 1p build is happy. FYI: this is the suggested solution from absl team.